### DISCUSSION TITLE
Dialogs

### DISCUSSION THREAD

**Discussion by alice-ang** [GECO-Project-App/facilitation-app#60] [3 - Design]
I feel like we currently have a lot of repeating pages with similar content that could be replaces with dialogs or something similar. Two pages I think are well suited to be converted to dialogs are the landing page for exercises and the accomplishment page.  The accomplishment page might be better suited since you may want to be able to link someone to the landing page of an exercise? 

Here's some some components shadcn offers that might be of use 
https://ui.shadcn.com/docs/components/drawer
https://ui.shadcn.com/docs/components/dialog
https://ui.shadcn.com/docs/components/sheet

<img width="642" alt="Screenshot 2024-09-18 at 09 31 03" src="https://github.com/user-attachments/assets/ce46593b-bb4c-4ee5-96ad-b9ad7077b6f8">

**Comment by eliasfaltin** [comment]
Interesting idea: I think there are upsides and downsides. Here are my thoughts. Let's keep discussing!

- Success pages could be a Modal (Dialog as it is called within Shadcn). 
- The landing pages could be a sheet (drawers make more sense on Desktop since they open from the side typically).


The risk with both of these is that you:
1. can land in "modal hell," where everything triggers a modal, and you need to consider the edge cases of what happens if a user dismisses the modal by clicking outside of it
2. you open yourself up to getting into "nested modals" (a modal that opens another modal). That is a cardinal sin in UI, in my opinion. It's not a big issue for this use case, but once you start using modals, you open the bottle, and it might be hard to get the djin back in.
3. The most significant issue: I would find it tricky to determine which page the sheet or modal should open on top of:

For the exercise landing page sheet:
- does this open on top of the catalog page? 
- What happens when/if we allow users to build agendas consisting of multiple exercises
- will we get more space than in the exercise card this way so we can display slightly more text to explain the exercise?

For the success modal (dialog):
- Does this open on top of the last page of the exercise or the exercise catalog page?
- Can we prohibit users from closing the modal when tapping/clicking outside of it (which is the expected functionality for a modal, in my opinion)?
We can link from there to the feedback form (which has modal potential -> we go straight to modal hell), but the more I think about it, we should not have a button on that page: 
- A cleaner solution would be to play a "cutscene" one-shot animation (full screen) that auto-dismisses once it is finished playing (and moves the user to the exercise catalog or feedback page). Clicking anywhere on the screen during this "cutscene" dismisses the animation/cutscene and moves the user directly to the feedback/exercise catalog page.


@aripuguh @Cristian-Nass, what are your thoughts on this?

**Reply by aripuguh** [reply]
Great suggestion Alice! I agree with Elias, I like the drawer & dialog (modal) for the landing page & accomplished page, but as he said there might be more tricky issues that will occur together with those. I like the suggestion point from Elias: 

> A cleaner solution would be to play a "cutscene" one-shot animation (full screen) that auto-dismisses once it is finished playing (and moves the user to the exercise catalog or feedback page). Clicking anywhere on the screen during this "cutscene" dismisses the animation/cutscene and moves the user directly to the feedback/exercise catalog page.

and considering the users & the demographic, having nested models can be a pain in the ass for the user I imagine, especially if the interactive buttons, close/back button are tiny/hard/invisible to access, but I am up for the challenge if we decided to go with drawer & dialog component! https://uxmovement.com/forms/how-to-simplify-4-nested-modals-into-only-1/#:~:text=Managing%20multiple%20windows%20on%20a,operations%20into%20their%20own%20windows. 

Thank you.

**Comment by eliasfaltin** [comment]
Update:
We are now using a full-screen toast for the "thank you for providing feedback" page.

In the coming week, we will also convert the "great job" page used in the Start Stop Continue exercise to a toast that:
- can be dismissed by the user by tapping anywhere
- is auto-dismissed after 2 seconds (if the user doesn't tap anywhere within 2 seconds)
